### PR TITLE
Tasks: add watch mode

### PR DESCRIPTION
**Pull request by krampstudio**
```
pnpm install
cd pacakges/tasks
pnpm run watch
```

cc @coyotte508 when you're back

### CODE REVIEW

File: packages/tasks/package.json [hunk 1]
```diff
@@ -25,6 +25,7 @@
 		"format:check": "prettier --check .",
 		"prepublishOnly": "pnpm run inference-codegen && git diff --name-only --exit-code src && pnpm run build",
 		"build": "tsup src/index.ts --format cjs,esm --clean && tsc --emitDeclarationOnly --declaration",
+		"watch": "tsup src/index.ts --format cjs,esm --clean --watch",
```
Comment (julien-c) [inline packages/tasks/package.json:28]: there's no need for the `tsc --declaration` part of the watch command?

**Review by julien-c — APPROVED** [review]
Thanks, this is super useful

File: packages/tasks/package.json (hunk 1 above)
Comment (krampstudio) [reply packages/tasks/package.json:28]: oh, yes, I thought it was only for type checking but it generates the declarations map too. Fixing it

File: packages/tasks/package.json [hunk 2]
```diff
@@ -25,6 +25,9 @@
 		"format:check": "prettier --check .",
 		"prepublishOnly": "pnpm run inference-codegen && git diff --name-only --exit-code src && pnpm run build",
 		"build": "tsup src/index.ts --format cjs,esm --clean && tsc --emitDeclarationOnly --declaration",
+		"watch:export": "tsup src/index.ts --format cjs,esm --watch",
+		"watch:types": "tsc --emitDeclarationOnly --declaration --watch",
+		"watch": "npm-run-all --parallel watch:export watch:types",
```
Comment (julien-c) [inline packages/tasks/package.json:30]: parallel is fine, or should we first tsup, then tsc (i.e. sequentially)?

(i don't know enough to know)

Comment (krampstudio) [reply packages/tasks/package.json:30]: the process won't exit in watch mode, so only the 1st process will run in sequential mode.

**Pull request merged**